src/lwc/geListView/geListView.js

```diff
@@ -559,39 +600,54 @@ export default class geListView extends LightningElement {
     * 'LastModifiedDate' need to get parsed/reassigned so we can display more user
     * friendly values i.e. Name rather than RecordId or a formatted date.
     *
-    * @param {list} dataRecords: List of sObject records
+    * @param {array} dataRecords: List of sObject records
     */
-    setDatatableRecordsForImperativeCall(dataRecords) {
+     setDatatableRecordsForImperativeCall(dataRecords) {
         this.records = [];
-        let recordUrl = this.getRecordUrl();
 
         let records = deepClone(dataRecords);
         records.forEach(record => {
             Object.keys(record).forEach(key => {
-                if (record[key].Name) {
+                let fieldDescribe = this.objectInfo.fields[key];
+
+                if (isNotEmpty(fieldDescribe)
+                    && (fieldDescribe.nameField ||
```

Comment:
What do you think about turning the second condition here into a constant and referencing that? Maybe something like `const isReferenceField = fieldDescribe.nameField || fieldDescribe.reference;`

Reply:
yeah sounds good. might help make it a little more clear what we're checking for there. done!

Reply:
@psadie-sfdo while I was testing through this I found that I think we'll need to keep the for loop syntax the way it was since `fieldDescribe` can potentially be undefined if the field isn't part of the `objectInfo` property, and we have several of those extra fields which are in the object keys.